pre_commit_hooks/string_fixer.py

```diff
@@ -32,7 +33,14 @@ def get_line_offsets_by_line_no(src):
 
 
 def fix_strings(filename):
-    contents = io.open(filename).read()
+    try:
+        contents = io.open(filename).read()
+    except Exception:
```

Comment:
I tried my hand at this but ran into some issues. I'll keep playing with it at some point, but I'll post what I've come across already here:

We definitely need to read the file as bytes:

    contents = io.open(filename, 'rb').read()

That breaks casting to a list, but fortunately there's a byte-equivalent for that

    splitcontents = bytearray(contents)

That also means we need to tokenize differently; fortunately, `tokenize.tokenize` handles bytes:

    tokens = reversed(list(tokenize.tokenize(io.BytesIO(contents).readline, )))

When setting the new characters in the loop, we need to modify a `bytearray` now instead of a `list`. That'll be _something along the lines of_:

    splitcontents[...] = int.from_bytes(new_text.encode(), byteorder='big')

Finally, we can set

    new_contents = b''.join(splitcontents)

The above runs into a problem where somewhere along the lines I f*cked up and started tracking characters instead of lines (or something like that? In progress, still needs work, if anyone has any interest in carrying on with this _be my guest_).